### PR TITLE
Stop trying after password found

### LINKED ISSUE #7: Doesn't stop on first match
The program keeps on trying all the passwords until the wordlist file ends or until the pipe is closed when the input ceases. This is an issue especially because

1. It wastes time, trying passwords that do not need to be tried
2. It may never terminate if the generator used does not terminate on it's own (for example hashcat with stdout in brute force mode)

Test case

```bash
echo "This is a secret" > secret.txt
ccrypt -e -K p4ssw0rd secret.txt
cptcrypt-ng -w /usr/share/seclists//Passwords/Leaked-Databases/rockyou-75.txt
```

This will run through all words in the list. (check against `wc -l`)

If you have a substancial wordlist and you put the test case at the very start (say the entire rockyou database and your password is 123456), then it will run through all 14M passwords.

```text
cptcracker-ng -w [...]/SecLists/Passwords/Leaked-Databases/rockyou.txt -f testcrypt.txt.cpt

 ~ CPT Cracker-ng v2.0.0-dev ~
 - File......: testcrypt.txt.cpt
 - Generator.: rockyou.txt
 . Worked at 421,948 pwd/sec
   Combinations: 14,346,259
   Working time: 34 sec
 + Password found: p4ssw0rd
   HEXA[ 70 34 73 73 77 30 72 64 ]
 ^ Ex(c)iting.
                                                                                                                                                                   [ 1m-25,898s ]
grep -n "p4ssw0rd" [...]/SecLists/Passwords/Leaked-Databases/rockyou.txt
40860:p4ssw0rd
[...]
wc -l [...]SecLists/Passwords/Leaked-Databases/rockyou.txt
14344391[...]/SecLists/Passwords/Leaked-Databases/rockyou.txt
```

With 422k pwd/s it shouldn't take 34 seconds to reach the 41000th line but rather about 1 second.

### PR DESCRIPTION
**Pull request by FalcoGer**
Fixes #7

### CODE REVIEW

File: src/shared/cracker.cc [hunk 1]
```diff
@@ -694,6 +694,7 @@ void Cracker::crack() {
 			}
 		}
 #endif
+		if (!this_is_now_we_fight) break; // we found a match, we can stop trying.
```
Comment (BoboTiG) [inline src/shared/cracker.cc]: Would it be better to put that condition in the `for` loop, at line 653?

Comment (FalcoGer) [reply src/shared/cracker.cc]: probably. perhaps make it a while loop instead.

Comment (BoboTiG) [reply src/shared/cracker.cc]: No, without changing the loop itself. Not tested, but maybe that could work?
```c
for ( ; (this_is_now_we_fight = this->cfgets(input, password, PWD_MAX)) ; this_is_now_we_fight ) {
```

Comment (FalcoGer) [reply src/shared/cracker.cc]: I just added another commit

**Comment by FalcoGer** [comment]
I tested with the test case above as well as empty file as wordlist and empty stdin (`echo -n ""`). it seems to work. again, didn't test zip anything.
My naming could've been better. it wasn't an endless for loop after all.

**Comment by BoboTiG** [comment]
Thanks a lot @FalcoGer ! :)

**Pull request merged**